docs/dev/architecture.md

```
@@ -0,0 +1,124 @@
+# Architecture Overview
+
+This document describes the high-level architecture of Tanjun.
+
+## Project Layout
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)